.npmrc

```diff
@@ -0,0 +1 @@
+package-lock=false
```

Comment:
What is the motivation? Why would you not want a `package-lock.json`?

In other projects, we're very happy to have repeatable builds thanks to having a `package-lock.json` artifact in the repository.

Reply:
Locking dependencies is good for apps, but not suitable for libraries, since it's not distributed with package. End-users and developer will end with different dependency tree, that could lead to bugs.

I have nice example. With one of the latest PR i got `mocha` failed to run tests, while it works perfectly on my local machine. After removing `package-lock.json` i was able to reproduce problem.

Reply:
Actually, I think best practices today are that you commit the package-lock.json (or yarn.lock, if using yarn) to the repository. You exclude it via .npmignore, and even if nunjucks is installed as a peer dependency directly from git, it won't look to the package-lock.json file. It is only used for development. So, if the package-lock.json file had been committed to git, you wouldn't have encountered the mocha issue.

Reply:
When one of the `dependencies` will get new version, you didn't get it locally, like package user do.

Even CI would not catch a bug, because it will resolve version from lock file, while package users will resolve new dependency version from `package.json`.

Lock file is bad idea for libraries. It good for apps, that should remain same.

Reply:
That is only true of the packages in `dependencies`, of which there are very few. By including lock files, we would be following the lead of the largest node projects: vue.js, vscode, React, Nest, Angular, next.js, and many more—all of whom have either a yarn.lock or a package-lock.json file committed to the repository.

Typically, adding a lock file to source control is done alongside integrating with something like dependabot. These bots open separate pull requests for every dependency update. This allows you to identify and fix bugs caused by upgrades (because they cause CI failures) _before_ they start showing up in unrelated pull requests of contributors.

Reply:
We use lock files in the Sinon projects, in order to have repeatable builds.

Reply:
I have remove this file. But i still believe that `package-lock.json` is redundant for a library. But i personally like what this guy wrote https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514